### PR TITLE
[AssetMapper] Search & filter assets in `debug:asset-mapper` command

### PR DESCRIPTION
**Pull request by smnandre**
| Q             | A
| ------------- | ---
| Branch?       | 7.2
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no 
| Issues        | Fix #... 
| License       | MIT

* Add `name` optional argument to filter paths & assets
* Add `--ext` option to filter assets by extension (e.g "css")
* Add `--vendor` option to only show assets from vendor packages (and `--no-vendor` to hide them)
* Update command help

Update:
~~* skip flappy test using chmod on Windows~~ (done by @xabbuh in #58152)

### CODE REVIEW

File: src/Symfony/Component/AssetMapper/Tests/ImportMap/RemotePackageStorageTest.php [hunk 1]
```diff
@@ -43,14 +43,14 @@ public function testGetStorageDir()
 
     public function testSaveThrowsWhenFailing()
     {
+        if ('\\' === \DIRECTORY_SEPARATOR) {
+            $this->markTestSkipped('This test cannot run on Windows.');
+        }
```
Comment (smnandre) [inline src/Symfony/Component/AssetMapper/Tests/ImportMap/RemotePackageStorageTest.php]: Still got errors (on AppVeyor) with this test, the "chmod" function cannot be relied on
... and this is the only way to trigger the expected exception

So let's skip it on Windows (i based this on Filesystem tests doing the same thing)

Comment (smnandre) [reply src/Symfony/Component/AssetMapper/Tests/ImportMap/RemotePackageStorageTest.php]: A new one (again the same problem i suppose of conflct between tests?)

Comment (smnandre) [reply src/Symfony/Component/AssetMapper/Tests/ImportMap/RemotePackageStorageTest.php]: Update: let's try again after #58152

**Comment by smnandre** [comment]
AssetMapper Tests are green on AppVeyor 🍾   
* (failures on [Validator](https://ci.appveyor.com/project/fabpot/symfony/builds/50526188#L2026) / intl)

PHPUnit Tests on 8.4 unrelated
* (failures on [Processs](https://github.com/symfony/symfony/actions/runs/10681905005/job/29606537729?pr=58141#step:8:2788) / pty)

Integration tests on 8.2 unrelated
* (failures on [Ldap](https://github.com/symfony/symfony/actions/runs/10681905001/job/29606536133?pr=58141#step:16:84))

**Comment by smnandre** [comment]
Ping someone for 7.2 ? 😊

File: src/Symfony/Component/AssetMapper/Command/DebugAssetMapperCommand.php [hunk 2]
```diff
@@ -40,10 +42,41 @@ public function __construct(
     protected function configure(): void
     {
         $this
+            ->addArgument('name', InputArgument::OPTIONAL, 'An asset name (or a path) to search for (e.g. "app")')
+            ->addOption('ext', null, InputArgument::OPTIONAL, 'Filter assets by extension (e.g. "css")', null, ['js', 'css', 'png'])
```
Comment (kbond) [inline src/Symfony/Component/AssetMapper/Command/DebugAssetMapperCommand.php]:
```suggestion
            ->addOption('ext', null, InputOption::VALUE_REQUIRED, 'Filter assets by extension (e.g. "css")', null, ['js', 'css', 'png'])
```

Just `--ext` is invalid, correct?

**Review by kbond — COMMENTED** [review]
This looks useful indeed!

File: src/Symfony/Component/AssetMapper/Command/DebugAssetMapperCommand.php (hunk 2 above)
Comment (smnandre) [reply src/Symfony/Component/AssetMapper/Command/DebugAssetMapperCommand.php]: Absolutely, thank you!

**Review by kbond — APPROVED** [review]

**Review by OskarStark — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @smnandre.

**Pull request merged**